File: android_fragment/app/src/main/java/com/support/generator/common/Presenter.java
package com.support.generator.common;

import io.reactivex.disposables.CompositeDisposable;

/*
 * Created by troy379 on 08.06.17.
 */
public abstract class Presenter<IVIEW> implements IPresenter<IVIEW> {

  protected IVIEW iview;
  protected CompositeDisposable disposables = new CompositeDisposable();

  public void onViewAttached(IVIEW iview, boolean created) {
    this.iview = iview;
  }

  public void onViewDetached() {
    this.iview = null;
  }

  public void onDestroy() {
    disposables.clear();
  }
}

File: example/app/src/main/java/gauges/mobile/github/com/mvpapplication/common/mvphelper/Presenter.kt
package gauges.mobile.github.com.mvpapplication.common.mvphelper

import io.reactivex.disposables.CompositeDisposable

/*
 * Created by troy379 on 08.06.17.
 */
abstract class Presenter<IVIEW> : IPresenter<IVIEW> {

    protected var iview: IVIEW? = null
    protected val disposables = CompositeDisposable()

    override fun onViewAttached(iview: IVIEW, created: Boolean) {
        this.iview = iview
    }

    override fun onViewDetached() {
        this.iview = null
    }

    override fun onDestroy() = disposables.clear()
}